vendor/github.com/checkpoint-restore/checkpointctl/lib/metadata.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I'm not sure we should add this dependency.

First, do we need to? It looks like the checkpoints are written in a format we can scan and sort by time... is there a reason not to make the cleanup function glob, sort by time, and delete the oldest N checkpoints?

Second, this adds a dependency that has a decent number of transitive dependencies, which usually only increase over time ... https://github.com/checkpoint-restore/checkpointctl/blob/main/go.mod

Third, this makes us depend on a persisted file controlled by an external library, which the compatibility guarantees around are unclear.

Reply:
It is a single file I use to have common checkpoint/restore definitions across Podman, CRI-O (and maybe containerd if my PR to support CRI checkpointing gets merged). I can define it locally if that helps. I just tried to define everything around checkpoint/restore metadata there. At this point it does not add additional dependencies to Kubernetes and I am trying to keep it that way.

Reply:
> First, do we need to? It looks like the checkpoints are written in a format we can scan and sort by time... is there a reason not to make the cleanup function glob, sort by time, and delete the oldest N checkpoints?

I explicitly did not want to do that. I wanted to avoid to walk the directory every time and compare file names and `stat()` each file. It felt unnecessary complicated when it is possible to control the metadata.

Reply:
An extra library and a persisted metadata file we have to worry about compatibility of, corruption of, cleanup of, etc, seems harder to reason about to me. I'd like input from kubelet owners on this aspect.